### DISCUSSION TITLE
iceoryx_management size

### DISCUSSION THREAD

**Discussion by forestf255** [eclipse-iceoryx/iceoryx#2012] [Q&A] [answered]
Hi all!

I've been experimenting with iceoryx and it seems like the size RouDi's allocated shmem [iceoryx_management] is quite high. Even when turning down all the options listed in the [configuration guide](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/configuration-guide.md) to 1, the reserved shared memory size is still ~19Mb. 

```
// Output from running iox-roudi...
Reserving 19781312 bytes in the shared memory [iceoryx_mgmt]
```

Does anyone have any ideas on why this amount of memory is needed? As far as I've read, this memory is separate from the mempools that are used for data transfer and only used for publisher/subscriber bookkeeping.

Posting my configuration if that is of any help - also I am testing on Linux. Another side note, max_publishers seems to need to be > 5 for iox-roudi to run, although this is a separate issue that is not particularly important to me.

```
-- [i] <<<<<<<<<<<<< Start iceoryx_posh configuration: >>>>>>>>>>>>>
-- [i] Using m:n communication!
-- [i] IOX_MAX_PUBLISHERS:6
-- [i] IOX_MAX_SUBSCRIBERS:1
-- [i] IOX_MAX_INTERFACE_NUMBER:1
-- [i] IOX_MAX_SUBSCRIBERS_PER_PUBLISHER:1
-- [i] IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY:1
-- [i] IOX_MAX_PUBLISHER_HISTORY:1
-- [i] IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY:1
-- [i] <<<<<<<<<<<<<< End iceoryx_posh configuration: >>>>>>>>>>>>>>
-- [i] <<<<<<<<<<<<< Start iceoryx_posh configuration: >>>>>>>>>>>>>
-- [i] Using m:n communication!
-- [i] IOX_MAX_PUBLISHERS:6
-- [i] IOX_MAX_SUBSCRIBERS:1
-- [i] IOX_MAX_INTERFACE_NUMBER:1
-- [i] IOX_MAX_SUBSCRIBERS_PER_PUBLISHER:1
-- [i] IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY:1
-- [i] IOX_MAX_PUBLISHER_HISTORY:1
-- [i] IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY:1
-- [i] <<<<<<<<<<<<<< End iceoryx_posh configuration: >>>>>>>>>>>>>>

       CMake Options
          CMAKE_BUILD_TYPE.....................: Release
          CMAKE_TOOLCHAIN_FILE.................:
          CMAKE_EXPORT_COMPILE_COMMANDS........: ON

       iceoryx Options
          BINDING_C............................: ON
          BUILD_ALL............................: OFF
          BUILD_DOC............................: OFF
          BUILD_SHARED_LIBS....................: ON
          BUILD_STRICT.........................: OFF
          BUILD_TEST...........................: OFF
          CCACHE...............................: ON (ccache not found)
          CLANG_TIDY...........................: OFF
          COVERAGE.............................: OFF
          DDS_GATEWAY..........................: OFF
          DOWNLOAD_TOML_LIB....................: ON
          EXAMPLES.............................: OFF
          INTROSPECTION........................: OFF
          ONE_TO_MANY_ONLY ....................: OFF
          ROUDI_ENVIRONMENT....................: OFF
          SANITIZE.............................: OFF
          TEST_WITH_ADDITIONAL_USER ...........: OFF
          TOML_CONFIG..........................: ON

       Build Properties
          project name..............: iceoryx_package
          c compiler................: /usr/bin/gcc-11
          c++ compiler..............: /usr/bin/g++
          cmake.....................: 3.22.1
```

**Comment by elBoberido** [comment, marked as answer]
@forestf255 without looking at the code I would guess this is related to the introspection. It has a separate mempool which currently cannot be configured. That's also the reason for the 5 publishers. I think the discovery feature also adds up to the memory. It should not be that complicated to add an option to disable all of the introspection and the discovery feature but would probably take at least one week of work to get everything right.

There is also a [small memory example](https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_examples/small_memory/README.md) but it essentially does what the configuration guide describes.

**Reply by forestf255** [reply]
Thanks! It looks like the small memory example introduces some new knobs to turn. I will check that out.
